src/actions/mod.rs

```diff
@@ -156,6 +156,23 @@ impl ActionHandler {
         }
     }
 
+    pub fn find_impls(&self, id: usize, params: TextDocumentPositionParams, out: &Output) {
+        let span = self.convert_pos_to_span(&params.text_document, params.position);
+        let type_id = self.analysis.id(&span).unwrap();
+        // FIXME send failure
+        let res = self.analysis
+            .find_impls(type_id)
+            .unwrap_or(vec![])
+            .into_iter()
+            // FIXME: these are part of libstd and libcore but the path probably does not
+            // exists and thus should be removed. Replace with correct path later.
+            .filter(|x| x.file.starts_with(Path::new("/checkout/src/")))
```

Comment:
I assume this is an artifact how the save-analysis data is generated. I assume this is the path used in the buildbots.

In general this can only become a problem for the values `ref_spans` and `impls` in `PerCreateAnalysis`, because here we map to a span. `ref_spans` is only filled by the `"refs":` array in the serialized json. `globs` and `def_id_for_span` map a span to something else. If they contain such a broken span this only leads to a failed lookup but not to wrong filenames being presented to the user.

In the local (`target/rls`) analysis data I can only find references to the path `/checkout/src/libcore/macros.rs`. They appear here only in the `macro_ref` section, which is currently unused by rls-analysis, thus they could not have been a problem. No span in `"refs"` contains such a path.

In the global ('rustlib/.../analysis') analysis data I see references to a lot of different files of the libraries. But none of those json files have any values in `"refs"`, which is the only existing possibly problematic case.

This mean, neither the local nor global data could have contributed until this change with broken spans.